From ad0b0475ab947ce3e15e14c4acff71f962b6ba10 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 23 Dec 2020 20:30:53 +0200
Subject: [PATCH] [SW-24746] preboot: set interrupt mask to default

Currently while running I2C slave in debug mode, it is observed
that the I2C slave interrupt mask is set to 0 in uboot shutdown sequence
which causes SCL low issue and reads from I2C slave debud mode doesn't
work at preboot level after LKD rmmod. As part of the hard reset,
PSOC block is not reset.

This patch add changes to set I2C slave interrupt mask to default value
as part of I2C slave init sequence. The setting of interrupt mask is
done platform specific as the mask values are different for different
platforms.

Change-Id: I359bcd59bb29eadd170889ff483835f16aa3855e
---

diff --git a/common/module_i2c_slave.c b/common/module_i2c_slave.c
index ae81361..fa04fae 100644
--- a/common/module_i2c_slave.c
+++ b/common/module_i2c_slave.c
@@ -66,6 +66,9 @@
 	/* Program IC_CON */
 	regmap_writel(0x22, mmPSOC_I2C_S_IC_CON);
 
+	/* Set INTR mask to default */
+	regmap_writel(I2C_SLAVE_DEFAULT_INTR_MASK, mmPSOC_I2C_S_IC_INTR_MASK);
+
 	/* Write to IC_RX_TL to set Rx FIFO threshold level */
 	regmap_writel(0x0, mmPSOC_I2C_S_IC_RX_TL);
 
diff --git a/gaudi/common.h b/gaudi/common.h
index 743d8a3..0b2f946 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -30,6 +30,7 @@
 #define FUSE_WORDS_PER_BANK		64
 #define I2C_SLAVE_ADDRESS		0x4f
 #define I2C_SLAVE_ADDRESS_DBG		0x47
+#define I2C_SLAVE_DEFAULT_INTR_MASK	0x8FF
 
 #define GAUDI_NICS_NUM			5
 #define GAUDI_NIC_ENGS_NUM		(GAUDI_NICS_NUM * 2)
diff --git a/goya/common.h b/goya/common.h
index b6cb1ad..2adf5b8 100644
--- a/goya/common.h
+++ b/goya/common.h
@@ -32,5 +32,6 @@
 #define FUSE_WORDS_PER_BANK	0x10
 #define I2C_SLAVE_ADDRESS	0x4f
 #define I2C_SLAVE_ADDRESS_DBG	0x47
+#define I2C_SLAVE_DEFAULT_INTR_MASK	0x8FF
 
 #endif /* COMMON_GOYA_H_ */
diff --git a/goya2/common.h b/goya2/common.h
index f9ed2ae..9ae9adb 100644
--- a/goya2/common.h
+++ b/goya2/common.h
@@ -36,6 +36,7 @@
 #define FUSE_WORDS_PER_BANK		0x40
 #define I2C_SLAVE_ADDRESS		0x6A
 #define I2C_SLAVE_ADDRESS_DBG		0x47
+#define I2C_SLAVE_DEFAULT_INTR_MASK	0x48FF
 
 #define ARC_RST_VALUE			0x110
 #define ARC_RST_VEC_ADDR		0x0
diff --git a/goya2_arc_gnu/common.h b/goya2_arc_gnu/common.h
index 336003d..63ce27f 100644
--- a/goya2_arc_gnu/common.h
+++ b/goya2_arc_gnu/common.h
@@ -33,6 +33,7 @@
 #define FUSE_WORDS_PER_BANK	0x40
 #define I2C_SLAVE_ADDRESS	0x6A
 #define I2C_SLAVE_ADDRESS_DBG	I2C_SLAVE_ADDRESS
+#define I2C_SLAVE_DEFAULT_INTR_MASK	0x48FF
 
 #define RECOVERY_TEXT_BASE	(0x0 + 0x4)
 #define RECOVERY_SIZE		0x4000
